khal/ui/widgets.py

```diff
@@ -424,3 +424,107 @@ def keypress(self, size, key):
             if not self._validate():
                 return
         return super().keypress(size, key)
+
+
+class DurationWidget(urwid.WidgetWrap):
+
+    @staticmethod
+    def unsigned_int(number):
+        """test if `number` can be converted to a positive int"""
+        try:
```

Comment:
Looks like my previous comment got lost in the void during rebasing.

IMHO, this looks simpler/cleaner:

```python
try:
    return int(number) >= 0
except ValueError:
    return False
```